Error compiling for ESP8266

I tried to compile this for an ESP8266 and get the following errors.

WARNING: library ThermistorLibrary-master claims to run on (avr) architecture(s) and may be incompatible with your current board which runs on (esp8266) architecture(s).
C:\Arduino\libraries\ThermistorLibrary-master\src\thermistor.cpp: In member function 'float thermistor::analog2temp()':

C:\Arduino\libraries\ThermistorLibrary-master\src\thermistor.cpp:41:38: error: ISO C++ forbids comparison between pointer and integer [-fpermissive]

     for (i = 1; i < heater_ttbl_map[e]; i++) {

Any sugestions as to how to fix this?

Otherwise the library works a treat on the NANO, however i need to report the temperature back via WiFi and the ESP seemed to be the easier option for this.

--- a/src/thermistor.cpp
+++ b/src/thermistor.cpp
@@ -38,7 +38,7 @@ float thermistor::analog2temp() {
     float celsius = 0;
     uint8_t i;
     short(*tt)[][2] = (short(*)[][2])(heater_ttbl_map[e]);
-    for (i = 1; i < heater_ttbl_map[e]; i++) {
+    for (i = 1; i < heater_ttbllen_map[e]; i++) {
       if (PGM_RD_W((*tt)[i][0]) > raw) {
         celsius = PGM_RD_W((*tt)[i - 1][1]) +
                   (raw - PGM_RD_W((*tt)[i - 1][0])) *

